src/releasers/go-yoshi.ts

```diff
@@ -0,0 +1,151 @@
+// * we need a deny list of modules that are mono repos, which we keep track of
+// ideally with minimal human intervention:
+//   * these should not show up in CHANGELOG for an "all" release.
+// * We need to also be able to manage the release of our snowflake, hand-written
+// libraries. Which get released.
+// * top level changes relate to all.
+
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {ReleasePR, ReleaseCandidate} from '../release-pr';
+
+import {ConventionalCommits} from '../conventional-commits';
+import {GitHubTag} from '../github';
+import {checkpoint, CheckpointType} from '../util/checkpoint';
+import {Update} from '../updaters/update';
+import {Commit} from '../graphql-to-commits';
+
+import {ReleaseType} from 'semver';
+import * as semver from 'semver';
+
+// Generic
+import {Changelog} from '../updaters/changelog';
+
+// These repos should be ignored when generating a release for the
+// top level mono-repo:
+const IGNORED_PATHS = [
+  'bigtable',
+  'bigquery',
+  'datastore',
+  'firestore',
+  'logging',
+  'pubsub',
+  'spanner',
+  'storage',
+];
+const GAPIC_PR_REGEX = /.*auto-regenerate gapics.*/;
+
+export class GoYoshi extends ReleasePR {
+  static releaserName = 'go-yoshi';
+  protected async _run() {
+    const latestTag: GitHubTag | undefined = await this.gh.latestTag(
+      this.monorepoTags ? `${this.packageName}-` : undefined
+    );
+    let gapicPR: Commit;
+    const commits: Commit[] = (
+      await this.commits({
+        sha: latestTag ? latestTag.sha : undefined,
+        path: this.path,
+      })
+    ).filter(commit => {
+      for (const ignoredPath of IGNORED_PATHS) {
```

Comment:
A few more comments on what's happening through here for future generations would be lovely :)